definitions.h

```diff
@@ -240,16 +245,123 @@ class Attributes
         void * ptrImg;  // points to the .bmp
 
         std::vector<double> values; // vector to allow for as many as the user would like
+        // std::vector<void*> ptrImgs; // vector to allow for as many textures as the user would like
+
+        double matrix[4][4];
 
         // easier to add another value to the vector
         void add(const double value) { values.push_back(value); }
 
         // clears the vector for future use
         void reset() { values.clear(); }
 
-        // I'm confused about what the Attributes class needs to be able to do in the future,
-        //    does it need more vectors of other data types? Does it need methods to access it's
-        //    values instead of direct access? Are other data types needed?
+        // multiplies the matrix by a 4x4 matrix
+        void multiply (Attributes &attr, const double rhs[][4])
+        {
+            std::cout << "old: " << attr.matrix[0][0];
+            double newMatrix[4][4];
+
+            for(int i = 0; i < 4; i++)
+                for(int j = 0; j < 4; j++)
+                    for(int k = 0; k < 4; k++)
+                    {
+                        newMatrix[i][j] += attr.matrix[i][k] * rhs[k][j];
+                    }
+                    
+            for(int i = 0; i < 4; i++)
+                for(int j = 0; j < 4; j++)
+                {
+                    attr.matrix[i][j] = newMatrix[i][j];
+                }
+            std::cout << "new: " << attr.matrix[0][0] << std::endl;
+        }
+};
+
+/*
+/***************************************************
+ * MATRIX
+ * Holds a matrix and functions to manipulate it
+ *************************************************
+class Matrix
+{
+};
```

Comment:
Here is where you can have separate functions for translating, rotating, and scaling. You also will want to add *= operator as well as = operator.